src/client/datascience/ipywidgets/ipyWidgetScriptSource.ts

```diff
@@ -185,53 +173,16 @@ export class IPyWidgetScriptSource implements IInteractiveWindowListener, ILocal
             }
         }
     }
-
-    /**
-     * Send the widget script source for a specific widget module & version.
-     * This is a request made when a widget is certainly used in a notebook.
-     */
-    private async sendWidgetSource(moduleName?: string, moduleVersion: string = '*') {
-        // Standard widgets area already available, hence no need to look for them.
-        if (!moduleName || moduleName.startsWith('@jupyter')) {
-            return;
-        }
-        if (!this.notebook || !this.scriptProvider) {
-            this.pendingModuleRequests.set(moduleName, moduleVersion);
-            return;
-        }
-
-        let widgetSource: WidgetScriptSource = { moduleName };
-        try {
-            widgetSource = await this.scriptProvider.getWidgetScriptSource(moduleName, moduleVersion);
-        } catch (ex) {
-            traceError('Failed to get widget source due to an error', ex);
-            sendTelemetryEvent(Telemetry.HashedIPyWidgetScriptDiscoveryError);
-        } finally {
-            // Send to UI (even if there's an error) continues instead of hanging while waiting for a response.
-            this.postEmitter.fire({
-                message: IPyWidgetMessages.IPyWidgets_WidgetScriptSourceResponse,
-                payload: widgetSource
-            });
-        }
-    }
-    private async saveIdentity(args: INotebookIdentity) {
-        this.notebookIdentity = args.resource;
-        await this.initialize();
-    }
-
-    private async initialize() {
+    public async initialize() {
```

Comment:
Why is this public now?

Reply:
Used internally and externally?

Reply:
In the notebook situation it may be that the notebook has already been created by the time the script source is created. It needs to be reinitialized at this point. In the old code the webview gets created before the notebook and the initialize gets called when the notebook opens.